File: advent-of-code-2024/day1.py
l = []
r = []
with open('input') as f:
    lines = f.readlines()
    for line in lines:
        l.append(int(line.split()[0]))
        r.append(int(line.split()[1]))

part1 = 0
l.sort()
r.sort()
for i in range(0, len(l)):
    part1 += abs(l[i] - r[i])
print("part 1 : ", part1)


part2 = 0

for i in range(0, len(l)):
    part2 += l[i] * r.count(l[i])

print("part 2 : ", part2)

File: advent-of-code-2024/cpp/day1.cpp
# include <iostream>
# include <fstream>
# include <vector>
# include <unordered_map>

using namespace std;

int main(){
    ifstream file("../input");

    vector<int> l, r;
    int ln, rn;
    unordered_map<int, int> rc;

    while(file >> ln >> rn){
        l.push_back(ln);
        r.push_back(rn);

        rc[rn]++;
    }

    sort(l.begin(), l.end());
    sort(r.begin(), r.end());

    long long part1 = 0;
    long long part2 = 0;
    for(int i=0; i<l.size(); i++){
        part1 += abs(l[i]-r[i]);
        part2 += l[i] * rc[l[i]];
    }


    cout << "part1 : " << part1 << endl;
    cout << "part2 : " << part2 << endl;
}